Reference for stage 3 arraybuffer-transfer (#28381)
mdn/content: files/en-us/web/javascript/reference/global_objects/arraybuffer/transfertofixedlength/index.md

---
title: ArrayBuffer.prototype.transferToFixedLength()
slug: Web/JavaScript/Reference/Global_Objects/ArrayBuffer/transferToFixedLength

> page-type: javascript-instance-method
> browser-compat: javascript.builtins.ArrayBuffer.transferToFixedLength

Reviewer: ```suggestion
page-type: javascript-instance-method
status:
  - experimental
browser-compat: javascript.builtins.ArrayBuffer.transferToFixedLength
```

---

> {{JSRef}}

Author: ```suggestion
{{JSRef}} {{SeeCompatTable}}
```

The **`transferToFixedLength()`** method of {{jsxref("ArrayBuffer")}} instances creates a new non-resizable `ArrayBuffer` with the same byte content as this buffer, then detaches this buffer.

## Syntax

```js-nolint
transferToFixedLength()
transferToFixedLength(newByteLength)
```

### Parameters

- `newByteLength`
  - : The {{jsxref("ArrayBuffer/byteLength", "byteLength")}} of the new `ArrayBuffer`. Defaults to the `byteLength` of this `ArrayBuffer`.

>     - If `newByteLength` is smaller than this `ArrayBuffer`'s `byteLength`, the rightmost bytes are dropped.

Author: ```suggestion
    - If `newByteLength` is smaller than the `byteLength` of this `ArrayBuffer`, the "overflowing" bytes are dropped.
```

>     - If `newByteLength` is larger than this `ArrayBuffer`'s `byteLength`, the rightmost bytes are filled with zeros.

Author: ```suggestion
    - If `newByteLength` is larger than the `byteLength` of this `ArrayBuffer`, the extra bytes are filled with zeros.
```

### Return value

A new {{jsxref("ArrayBuffer")}} object. Its contents are initialized to the contents of this `ArrayBuffer`, and extra bytes, if any, are filled with zeros. The new `ArrayBuffer` is always non-resizable. The original `ArrayBuffer` is detached.

### Exceptions

- {{jsxref("TypeError")}}
  - : Thrown if this `ArrayBuffer` is already detached.

## Description

> Unlike {{jsxref("ArrayBuffer/transfer", "transfer()")}}, `transferToFixedLength()` always creates a non-resizable `ArrayBuffer`. This means `newByteLength` can be larger than the `maxByteLength`, even if this `ArrayBuffer` is resizable.

Author: ```suggestion
Unlike {{jsxref("ArrayBuffer/transfer", "transfer()")}}, `transferToFixedLength()` always creates a non-resizable `ArrayBuffer`. This means `newByteLength` can be larger than the `maxByteLength`, even if this `ArrayBuffer` is resizable. See [transferring ArrayBuffers](/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer#transferring_arraybuffers) for more information.
```
